website/content/docs/k8s/installation/vault/data-integration/webhook-certs.mdx

```diff
@@ -0,0 +1,261 @@
+    max_ttl="720h"
+```
+
+Create a Vault role for the connect inject PKI engine will set the default certificate issuance parameters:
+
+```shell-session
+$ vault write connect-inject/roles/connect-inject-role \
+    allowed_domains="<Allowed-domains-string>" \
+    allow_subdomains=true \
+    allow_bare_domains=true \
+    allow_localhost=true \
+    generate_lease=true \
+    max_ttl="720h"
+```
+
+To generate the `<Allowed-domains-string>` for each use the following script as a template:
+
+```shell-session
+#!/bin/sh
+
+# NAME is set to either the value from `global.name` from your Consul K8s value file, or your $HELM_RELEASE_NAME-consul
+export NAME=consulk8s
+# NAMESPACE is where the Consul on Kubernetes is installed
+export NAMESPACE=consul
+# DATACENTER is the value of `global.datacenter` from your Helm values config file
+export DATACENTER=dc1
+
+echo allowed_domains_controller=\"${NAME}-controller-webhook,${NAME}-controller-webhook.${NAMESPACE},${NAME}-controller-webhook.${NAMESPACE}.svc,${NAME}-controller-webhook.${NAMESPACE}.svc.cluster.local\""
+
+echo allowed_domains_connect_inject=\"${NAME}-connect-injector,${NAME}-connect-injector.${NAMESPACE},${NAME}-connect-injector.${NAMESPACE}.svc,${NAME}-connect-injector.${NAMESPACE}.svc.cluster.local\""
+```
+
+### Create a Vault auth roles that link the policy to each Consul on Kubernetes service account that requires access
+
+-> **Note:** The Vault auth roles below will be your `global.secretsBackend.vault.controllerRole` and `global.secretsBackend.vault.connectInjectRole` Helm values respectively.
+
+
+Finally, Kubernetes auth roles need to be created for controller and connect inject webhooks.
+
+Role for Consul controller webhooks:
+```shell-session
+$ vault write auth/kubernetes/role/controller-role \
+    bound_service_account_names=<Consul controller service account> \
+    bound_service_account_namespaces=<Consul installation namespace> \
+    policies=controller-ca-policy \
+    ttl=1h
+```
+
+To find out the service account name of the Consul controller,
+you can run:
+
+```shell-session
+ $ helm template --release-name ${RELEASE_NAME} --show-only templates/controller-serviceaccount.yaml hashicorp/consul
+```
+
+Role for Consul connect inject webhooks:
+
+```shell-session
+$ vault write auth/kubernetes/role/connect-inject-role \
+    bound_service_account_names=<Consul connect inject service account> \
+    bound_service_account_namespaces=<Consul installation namespace> \
+    policies=connect-inject-ca-policy \
+    ttl=1h
+```
+
+To find out the service account name of the Consul connect inject, use the command below.
+```shell-session
+ $ helm template --release-name ${RELEASE_NAME} --show-only templates/connect-inject-serviceaccount.yaml hashicorp/consul
+```
+
+### Configure the Vault Kubernetes auth role in the Consul on Kubernetes helm chart
+
+Now that we've configured Vault, you can configure the Consul Helm chart to
+use the Server TLS certificates from Vault:
```

Comment:
```suggestion
### Update the Consul on Kubernetes helm chart

Configure the Consul Helm chart to use the Server TLS certificates from Vault. 
```
Is that cleaner?

Reply:
Yes.  Thank you.  I'll replace it on several pages.